ocaml/lambda/translclass.ml

```diff
@@ -208,7 +208,7 @@ let rec build_object_init ~scopes cl_table obj params inh_init obj_init cl =
            Typeopt.layout pat.pat_env pat.pat_loc pat.pat_type
          in
          Lambda.lfunction
-                   ~kind:(Curried {nlocal=0}) ~params:((param, param_layout)::params)
+                   ~kind:(Curried {nlocal=0}) ~params:({name = param; layout = param_layout; attributes = Lambda.default_param_attribute; mode = alloc_heap } ::params)
```

Comment:
Since the same pattern occurs a lot in this file, you might want to define a function `heap_param = fun name layout -> { name; layout; attributes = Lambda.default_param_attribute; mode = alloc_heap }`